test/corpus/cte.txt

```diff
@@ -231,52 +231,52 @@ FROM top_cte;
 --------------------------------------------------------------------------------
 
 (program
- (statement
-  (keyword_with)
-  (cte
-   (identifier)
-   (keyword_as)
-   (statement
+  (statement
```

Comment:
missing a possible test for cte in parentheses -- this parses in Postgres:

```sql
((((with x as ((((select * from ints)))) ((((select * from x))))))));
```

Reply:
I added a test for multiple layers of parenthesis but an arbitrary number of parenthesis might require a custom scanner.

Reply:
dollar quotes (beyond the few common types I added explicitly) need a custom scanner too, but I didn't have much luck with that when I tried a while back